EmpireLegends/GameStart.cpp

```diff
@@ -92,3 +92,30 @@ void GameStart::detPlayerCount() {
 
 	std::cout << "\nEach player gets " << playerCoins << " coins." << std::endl;
 }
+
+void GameStart::startGame()
+{
+	state = "WELCOME TO EMPIRE LEGENDS. THE GAME IS ABOUT TO START.";
+	Notify();
+}
+
+GameStartObserver::GameStartObserver(GameStart* s)
+{
+	subject = s;
```

Comment:
not sure why the observer would keep an instance of its subject? Isn't the goal of an observer instance to be held by the subject, and not the opposite?

Reply:
Nope this is how the pattern is implemented :)
https://www.tutorialspoint.com/design_pattern/observer_pattern.htm (also in slides)